lms/djangoapps/lti_provider/outcomes.py

```
@@ -0,0 +1,199 @@
+    """
+    Create the XML document that contains the new score to be sent to the LTI
+    consumer. The format of this message is defined in the LTI 1.1 spec.
+    """
+    envelope = etree.Element(
+        'imsx_POXEnvelopeRequest',
+        xmlns='http://www.imsglobal.org/services/ltiv1p1/xsd/imsoms_v1p0'
+    )
+    header = etree.SubElement(envelope, 'imsx_POXHeader')
+    header_info = etree.SubElement(header, 'imsx_POXRequestHeaderInfo')
+    version = etree.SubElement(header_info, 'imsx_version')
+    message_id = etree.SubElement(header_info, 'imsx_messageIdentifier')
+    body = etree.SubElement(envelope, 'imsx_POXBody')
+    replace_result_request = etree.SubElement(body, 'replaceResultRequest')
+    result_record = etree.SubElement(replace_result_request, 'resultRecord')
+    sourced_guid = etree.SubElement(result_record, 'sourcedGUID')
+    sourced_id = etree.SubElement(sourced_guid, 'sourcedId')
+    result = etree.SubElement(result_record, 'result')
+    result_score = etree.SubElement(result, 'resultScore')
+    language = etree.SubElement(result_score, 'language')
+    text_string = etree.SubElement(result_score, 'textString')
+
+    version.text = 'V1.0'
+    message_id.text = str(uuid.uuid4())
+    sourced_id.text = result_sourcedid
+    language.text = 'en'
+    text_string.text = str(score)
+
+    return etree.tostring(envelope, xml_declaration=True, encoding='UTF-8')
+
+
+def sign_and_send_replace_result(assignment, xml):
+    """
+    Take the XML document generated in generate_replace_result_xml, and sign it
+    with the consumer key and secret assigned to the consumer. Send the signed
+    message to the LTI consumer.
+    """
+    outcome_service = assignment.outcome_service
+    assert outcome_service, 'assignment.outcome_service guaranteed to be non-null'
+    consumer_key = outcome_service.consumer_key
+    assert consumer_key, 'outcome_service.consumer_key guaranteed to be non-null'
```

Perhaps this should throw an explicit exception?